Update pnp directions for samples (#1102)
Azure/azure-sdk-for-c: sdk/samples/iot/README.md

- *Executable:* `paho_iot_hub_pnp_sample`

  This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/paho_iot_hub_pnp_sample.c) connects an IoT Plug and Play enabled device (a thermostat) with the Digital Twin Model ID (DTMI) detailed [here](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json). If a timeout occurs while waiting for a message from the Azure IoT Explorer, the sample will continue. If 3 timeouts occur consecutively, the sample will disconnect. X509 self-certification is used.

  To interact with this sample, **you must use the Azure IoT Explorer**. The capabilities are listed below:

- **Device Twin**: Two device twin properties are supported in this sample:
  - A desired property named `targetTemperature` with a `double` value for the desired temperature.

>   - A reported property named `maxTempSinceLastReboot` with a `double` value for the highest temperature reached since boot.

Reviewer: ```suggestion
  - A reported property named `maxTempSinceLastReboot` with a `double` value for the highest temperature reached since device boot.
```

Reviewer: also, add a blank line after this.

  }
  ```

> - **Direct Method (Command)**: One method is supported in this sample: `getMaxMinReport`.

Reviewer: ```suggestion
- **Direct Method (Command)**: One device method is supported in this sample: `getMaxMinReport`.
```

- *Executable:* `paho_iot_hub_pnp_component_sample`

>   This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/paho_iot_hub_pnp_component_sample.c) extends the IoT Hub Plug and Play Sample above to mimic a Temperature Controller.  This sample connects the IoT Plug and Play enabled device (the Temerature Controller) with the Digital Twin Model ID (DTMI) detailed [here](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/TemperatureController.json).  If a timeout occurs while waiting for a message from the Azure IoT Explorer, the sample will continue. If 3 timeouts occur consecutively, the sample will disconnect. X509 self-certification is used.

Reviewer 2: ```suggestion
  This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/paho_iot_hub_pnp_component_sample.c) extends the IoT Hub Plug and Play Sample above to mimic a Temperature Controller.  This sample connects the IoT Plug and Play enabled device (the Temperature Controller) with the Digital Twin Model ID (DTMI) detailed [here](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/TemperatureController.json).  If a timeout occurs while waiting for a message from the Azure IoT Explorer, the sample will continue. If 3 timeouts occur consecutively, the sample will disconnect. X509 self-certification is used.
```

Author: I always appreciate your mad typo catching skills.  I am very good at creating typos lol.  Thank you.

Reviewer: don't put double spaces after periods.

Reviewer 2: > I always appreciate your mad typo catching skills. I am very good at creating typos lol. Thank you.

Other than a quick scan/read of docs, I use tools like VS Code and browser extensions. No skills needed :P

Author: @ericwol-msft thanks.  That is a typing style I was taught in grade school (25+ year habit), so I may need your help hammering this single-space style into me until I get it right :)

  This Temperature Controller is made up of the following sub-components:

  - Device Info
  - [Temperature Sensor 1](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json)

>   - [Temperature Sensor 2](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json)

Reviewer: yes

  To interact with this sample, **you must use the Azure IoT Explorer**. The capabilities are listed below:

> - **Device Twin**: 11 device twin properties are supported in this sample:

Reviewer: ```suggestion
- **Device Twin**: The following device twin properties are supported in this sample:
```

  "properties": {
      "desired": {
          "thermostat1": {
              "targetTemperature": 34.8
          },
          "thermostat2": {
              "targetTemperature": 68.5
          }
      }
  }
  ```

  Upon receiving a desired property message, the sample will update the twin property locally and
  send a reported property of the same name back to the service. This message will include a set of
  "ack" values: `ac` for the HTTP-like ack code, `av` for ack version of the property, and an
  optional `ad` for an ack description.

  ```json
  "properties": {
      "reported": {
          "thermostat1": {
              "__t": "c",
              "maxTempSinceLastReboot": 38.2,
              "targetTemperature": {
                  "value": 34.8,
                  "ac": 200,
                  "av": 27,
                  "ad": "success"
              }
          },
          "thermostat2": {
              "__t": "c",
              "maxTempSinceLastReboot": 69.1,
              "targetTemperature": {
                  "value": 68.5,
                  "ac": 200,
                  "av": 28,
                  "ad": "success"
              },
          }
      }
  }
  ```

> - **Direct Method (Command)**: Two methods are supported in this sample: `reboot` and `getMaxMinReport`.

Reviewer: ```suggestion
- **Direct Method (Command)**: Two device methods are supported in this sample: `reboot` and `getMaxMinReport`.
```

  If any other methods are attempted to be invoked, the log will report the method is not found. To invoke a method, select your device's Direct Method tab in the Azure IoT Explorer.

  - To invoke `reboot` on the Temperature Controller, enter the method name `reboot`. Select Invoke method.
  - To invoke `getMaxMinReport` on Temperature Sensor 1, enter the method name `thermostat1/getMaxMinReport` along with a payload using an [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) time format. Select Invoke method.
  - To invoke `getMaxMinReport` on Temperature Sensor 2, enter the method name `thermostat2/getMaxMinReport` along with a payload using an [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) time format. Select Invoke method.

  ```json
  "2020-08-18T17:09:29-0700"
  ```

  The command will send back to the service a response containing the following JSON payload with updated values in each field:

  ```json
    {
      "maxTemp": 74.3,
      "minTemp": 65.2,
      "avgTemp": 68.79,
      "startTime": "2020-08-18T17:09:29-0700",
      "endTime": "2020-08-18T17:24:32-0700"
    }
  ```

> - **Telemetry**: The Temperature Controller sends a JSON message with the property name `workingSet` and a `double` value for the current working set of the device memory in KiB.  Each Temperature Sensor sends a JSON message with the property name `temperature` and a `double` value for the current temperature.

Reviewer: ```suggestion
- **Telemetry**: The Temperature Controller sends a JSON message with the property name `workingSet` and a `double` value for the current working set of the device memory in KiB.  Also, each Temperature Sensor sends a JSON message with the property name `temperature` and a `double` value for the current temperature.
```
